src/averageAges.js

```diff
@@ -1,61 +1,48 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const man = people.filter(person => (century) ? (
+    person.sex === 'm' && Math.ceil(person.died / 100) === century)
+    : person.sex === 'm');
+
+  const sum = man.reduce(function(accumulator, currentValue) {
+    return accumulator + (currentValue.died - currentValue.born);
+  }, 0);
```

Comment:
You are using similar functions for men and women. So you can make one for both options.